djvm/build.gradle

```diff
@@ -69,10 +69,14 @@ shadowJar {
     exclude 'sandbox/java/lang/StackTraceElement.class'
     exclude 'sandbox/java/lang/StringBuffer.class'
     exclude 'sandbox/java/lang/StringBuilder.class'
+    exclude 'sandbox/java/lang/ref/ReferenceQueue.class'
```

Comment:
I am not understanding the "bootstrap" stage, where are they used? Why do you need to "compile" sandboxed classes rather than generating them at runtime?

Reply:
Some classes (i.e. the "template" ones) are copied into the sandbox rather than being generated from the originals in `rt.jar`. This is because the ones in `rt.jar` would be completely unsuitable. E.g. `sandbox.java.lang.Object` and `sandbox.java.lang.System`.

Reply:
That make sense. Although i am still confused by the role of "dummy" (stub) classes below, what base library class is compiled based on them? Is it for `ConcurrentHashMap` ?

Reply:
The full list of template classes (i.e. the ones that actually matter) is provided by `AnalysisConfiguration.TEMPLATE_CLASSES`.